starlite/cache/redis_cache_backend.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, Optional
+
+from pydantic import BaseModel
+
+from starlite import MissingDependencyException
+
+try:
+    from redis.asyncio import Redis
+    from redis.asyncio.connection import ConnectionPool
```

Comment:
What the difference for us between using redis and hiredis?

Reply:
`aioredis` which is the package that provides asyncio support for Redis was integrated into `redis-py` in version `4.2.0`:
https://github.com/redis/redis-py/releases/tag/v4.2.0rc1

[`aioredis` can be used with `hiredis`](https://aioredis.readthedocs.io/en/latest/getting-started/#with-hiredis) and so as [`redis-py`](https://github.com/redis/redis-py#installation) but I'm not sure if `redis-py` asyncio works with `hiredis`. Maybe we can just add `redis[hiredis]` instead or `redis`? 🤔

Reply:
there is a Hiredis parser in their asyncio module: https://github.com/redis/redis-py/blob/b0883b791f95a595fae70bcedf3ad0f73c00e258/redis/asyncio/connection.py#L412

Reply:
so I guess we should install `redis` with `hiredis` then, I'll make the fix

Reply:
Or should we just let people install hiredis themselves if they want the speedup?

Reply:
both options can work, whatever you guys prefer

Reply:
I'd make it optional but support it

Reply:
Done!